File: samples/Sample.ts
interface Color {
    r: number;
    g: number;
    b: number;
    toHex(): string;
}

type ColorMap = Record<string, Color>;

const MAX_RETRIES: number = 3;
const VERSION: string = "1.0.0";

class ThemeColor implements Color {
    constructor(
        public r: number,
        public g: number,
        public b: number
    ) {}

    toHex(): string {
        return `#${this.r.toString(16).padStart(2, "0")}` +
               `${this.g.toString(16).padStart(2, "0")}` +
               `${this.b.toString(16).padStart(2, "0")}`;
    }

    static fromHex(hex: string): ThemeColor {
        const h = hex.replace("#", "");
        return new ThemeColor(
            parseInt(h.substring(0, 2), 16),
            parseInt(h.substring(2, 4), 16),
            parseInt(h.substring(4, 6), 16)
        );
    }
}

class Theme {
    private colors: ColorMap = {};
    public readonly isDark: boolean = true;

    constructor(public name: string) {}

    addColor(key: string, color: Color): void {
        this.colors[key] = color;
    }

    getColor(key: string): Color | undefined {
        return this.colors[key];
    }
}

async function fetchData(url: string): Promise<unknown> {
    try {
        const response: Response = await fetch(url);
        if (!response.ok) {
            throw new Error(`HTTP ${response.status}: ${response.statusText}`);
        }
        return await response.json();
    } catch (error) {
        console.error("Fetch failed:", (error as Error).message);
        return null;
    }
}

function processItems<T>(items: T[], prefix: string = "item"): string[] {
    return items.map((item, i) => `${prefix}_${i}: ${item}`);
}

// Type guard
function isColor(value: unknown): value is Color {
    return typeof value === "object" && value !== null &&
        "r" in value && "g" in value && "b" in value;
}

const theme = new Theme("Islands Dark");
theme.addColor("keyword", ThemeColor.fromHex("#CF8E6D"));
theme.addColor("background", new ThemeColor(25, 26, 28));

const items = processItems(["alpha", "beta", "gamma"]);
export { Theme, ThemeColor, fetchData, processItems };

File: samples/Sample.go
package main

import (
	"fmt"
	"net/http"
	"strings"
	"time"
)

const (
	MaxRetries    = 3
	DefaultPort   = 8080
	Version       = "1.0.0"
	Pi            = 3.14159
)

type Color struct {
	R, G, B int
}

func (c Color) ToHex() string {
	return fmt.Sprintf("#%02X%02X%02X", c.R, c.G, c.B)
}

func NewColor(r, g, b int) *Color {
	return &Color{R: r, G: g, B: b}
}

type Theme struct {
	Name   string
	Colors map[string]Color
	Dark   bool
}

func NewTheme(name string) *Theme {
	return &Theme{
		Name: name,
		Colors: map[string]Color{
			"background": {R: 25, G: 26, B: 28},
			"foreground": {R: 188, G: 190, B: 196},
			"keyword":    {R: 207, G: 142, B: 109},
		},
		Dark: true,
	}
}

func (t *Theme) GetColor(key string) (Color, bool) {
	color, ok := t.Colors[key]
	return color, ok
}

func (t *Theme) Apply() error {
	for key, color := range t.Colors {
		fmt.Printf("  %s: %s\n", key, color.ToHex())
	}
	return nil
}

func fetchData(url string) ([]byte, error) {
	client := &http.Client{Timeout: 30 * time.Second}
	resp, err := client.Get(url)
	if err != nil {
		return nil, fmt.Errorf("fetch failed: %w", err)
	}
	defer resp.Body.Close()

	// Process the response
	return nil, nil
}

func main() {
	theme := NewTheme("Islands Dark")
	fmt.Printf("Theme: %s (dark: %v)\n", theme.Name, theme.Dark)
	fmt.Println(strings.Repeat("-", 40))
	theme.Apply()

	items := []string{"alpha", "beta", "gamma"}
	for i, item := range items {
		fmt.Printf("%d: %s\n", i, item)
	}
}